### PR TITLE
Enforce signed images

### PR DESCRIPTION
**Pull request by karthicgit**
To enforce signed images in OKE.

### CODE REVIEW

File: modules/oke/cluster.tf [hunk 1]
```diff
@@ -12,6 +12,13 @@ resource "oci_containerengine_cluster" "k8s_cluster" {
     subnet_id            = var.oke_cluster.cluster_subnets["cp"]
   }
 
+  image_policy_config {
+    is_policy_enabled = var.oke_cluster.use_signed_images
+    key_details {
+      kms_key_id = var.oke_cluster.use_signed_images == true ? var.oke_cluster.kms_key_id : null
```
Comment (hyder) [inline modules/oke/cluster.tf:18]: The key used for signed images can be different from the one used to encrypt etcd, so users must be able to specify each separately. 

Add 1 more variable to allow the user to specify the key for signed images and then pass it along to the oke module and eventually to this line of code.

**Review by hyder — CHANGES_REQUESTED** [review]

File: modules/oke/cluster.tf (hunk 1 above)
Comment (hyder) [reply modules/oke/cluster.tf:18]: Please add documentation for the new variable in terraform options as well.

Comment (karthicgit) [reply modules/oke/cluster.tf]: New variable signedimage_key_id has been added. Doc has been updated as well.

Comment (hyder) [reply modules/oke/cluster.tf]: 1. Please rename to image_signing_keys.
2. Sorry, I just noticed this: You can add multiple keys for image signing. So, this should be a list of key ids.
3. Add a dynamic and iterator so you can iterate over the list of key ids

Comment (karthicgit) [reply modules/oke/cluster.tf]: Looks like it does not support list. Inappropriate value for attribute "kms_key_id": string required.

Comment (hyder) [reply modules/oke/cluster.tf]: Try putting a dynamic at key_details level  and then add an iterator?

Comment (karthicgit) [reply modules/oke/cluster.tf]: Changes done please review. I have updated the default node_pool_os_version to 7.9 in tfvars.example as well.

**Review by hyder — APPROVED** [review]

**Review by hyder — DISMISSED** [review]
Approving review with incoming changes.

**Review by hyder — APPROVED** [review]

**Pull request merged**